### ISSUE TITLE
[cms-ci] CI DX Improvements

### ISSUE THREAD

**Issue by ElijahLynn** [department-of-veterans-affairs/va.gov-cms#1229]
## Developer Experience
- [x] https://github.com/opendevshop/devshop/issues/530 - Each PR “Status Check” details link points to the correct log
  - [x] Finish this feature in Provision-Ops/yaml-tests.
  - [x] Follow up: https://github.com/opendevshop/devshop/issues/528 - Finish migrating to DevShop/YamlTasks. 
  - [x] https://github.com/opendevshop/devshop/pull/549 Finish importing Power process into devshop.
  - [x] https://github.com/opendevshop/devshop/issues/552 Release new versions of devshop (which will now automatically release new versions of power-process and yaml-tasks!)
  - [x] #1304 Submit new PR with `yaml-tasks` instead of `yaml-tests`.
  - [x] #1303 Ensure new command works in devops repo: Search for any instances of `yaml-tests` or `composer yaml-tests`.
- [ ] https://github.com/opendevshop/devshop/issues/527 - Deployments section of a PR shows actual deployments. If this works well we may not need the comment item.  
    - [ ] Deployment API: https://github.com/opendevshop/devshop/deployments
    - pr123.web.ci.cms.va.gov 
    - pr123.ci.cms.va.gov 
- [ ] https://github.com/opendevshop/devshop/issues/540 - When PR Review Instance environment is built and ready, a link to the new environments is posted as a comment to the GitHub PR itself by the `va-cms-bot` user.  
  - [ ] TODO: Don't start until Deployments and Commit status API. integration tasks are done, to assess if we need to use comments at all.
e.g. https://github.com/department-of-veterans-affairs/va.gov-cms/pull/299#issuecomment-489758661

- [ ] https://github.com/provision-ops/yaml-tests/issues/54 - Fail Fast - @ElijahLynn 

# New Issues to create in open source repo.

- [ ] DevShop GitHub Actions Runner.
   - [ ] GitHub Actions Artifacts, etc.

**Comment by jonpugh** [comment]
## Implementation

- [ ] Yaml Tests 1.12:  Uses `--status-url` or `YAML_TASKS_STATUS_URL` environment variable as the url for the "Details" link next to each individual test: https://github.com/provision-ops/yaml-tests/releases/tag/v1.12.0
- [ ] DevShop PR#538: Add URL and ROOT environment vars to devshop_dothooks execution. https://github.com/opendevshop/devshop/pull/538
- [ ] Include Yaml-tests 1.12 in VA.gov-cms.
- [ ] Set --status-url option in tests.yml.

**Comment by jonpugh** [comment]
BLOCKED by update of provision-ops/power-process that jumps to symfony/process 4.

Drupal itself requires symfony 3 I think. 4 and 3 are not compatible, so we have to figure out a way to let Provision-ops/power-process utilize Symfony/process of one version while the rest can use another.

I wonder if composer version "aliasing" would work here? 

This is also related to the Drush 10 efforts. I don't think we can use symfony 4 components at all until Drupal 9 or some future 8.x.

**Comment by ElijahLynn** [comment]
Not sure if this should be in this epic but this would be good to do as well > https://github.com/provision-ops/yaml-tests/issues/54

**Comment by jonpugh** [comment]
Commit status links are working!  See #1304

![Screenshot from 2020-03-26 11-05-52](https://user-images.githubusercontent.com/106420/77662203-f0ba5980-6f51-11ea-9c0f-cc50b9091520.png)

TODO: 

- [ ] HTTPS Urls are required by GitHub. We should setup a global redirect from https to http urls on the devshop server.
- [ ] The commit status links now include an anchor for the specific test. We can make these work in devshop by altering the way `theme_devshop_ascii` formats links. It already turns URLs into links. The links are also printed in YamlTasks output. In order to get the #test/name anchors working, we need to parse the URLs output in `theme_devshop_ascii` from `https://x/node/123#anchor` to `<a href='https://x/node/123#anchor' id='yaml/test/name'`>

**Comment by jonpugh** [comment]
![Screenshot from 2020-03-26 11-32-08](https://user-images.githubusercontent.com/106420/77665090-8acfd100-6f55-11ea-8eda-eb21a96dd5a9.png)

**Comment by jonpugh** [comment]
Global HTTPS to HTTP redirect implemented:

This is a customization that should be put into the playbook, OR we will get a real cert for the devshop server.

```php
<?php # global settings.php
if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') {
  header("Location: http://{$_SERVER['HTTP_HOST']}/{$_SERVER['REQUEST_URI']}");
}
~

**Comment by ElijahLynn** [comment]
We are still finishing this, fwiw. Will re-open soon.
